File: CS/E5101/Models/PdfPageModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Drawing;
using System.IO;
using DevExpress.Pdf;
using System.Drawing.Imaging;

namespace E5101.Models {
    public class PdfPageModel {
        PdfDocumentProcessor _documentProcessor;

        public PdfPageModel(PdfDocumentProcessor documentProcessor) {
            this._documentProcessor = documentProcessor;
        }

        public PdfDocumentProcessor DocumentProcessor {
            get {
                return _documentProcessor;
            }
        }

        public int PageNumber {
            get;
            set;
        }

        public byte[] GetPageImageBytes() {
            using (Bitmap bitmap = DocumentProcessor.CreateBitmap(PageNumber, 900)) {
                using (MemoryStream stream = new MemoryStream()) {
                    bitmap.Save(stream, ImageFormat.Png);
                    return stream.ToArray();
                }
            }
        }
    }
}

File: VB/E5101/Models/PdfPageModel.vb
Imports System.Drawing
Imports System.IO
Imports DevExpress.Pdf
Imports System.Drawing.Imaging

Namespace E5101.Models

    Public Class PdfPageModel

        Private _documentProcessor As PdfDocumentProcessor

        Public Sub New(ByVal documentProcessor As PdfDocumentProcessor)
            _documentProcessor = documentProcessor
        End Sub

        Public ReadOnly Property DocumentProcessor As PdfDocumentProcessor
            Get
                Return _documentProcessor
            End Get
        End Property

        Public Property PageNumber As Integer

        Public Function GetPageImageBytes() As Byte()
            Using bitmap As Bitmap = DocumentProcessor.CreateBitmap(PageNumber, 900)
                Using stream As MemoryStream = New MemoryStream()
                    bitmap.Save(stream, ImageFormat.Png)
                    Return stream.ToArray()
                End Using
            End Using
        End Function
    End Class
End Namespace
